index.html

```diff
@@ -1,20 +1,450 @@
+      contains.
+    </p>
+    <p>
+      Implementations that use ECMAScript [[ECMA-262]] to implement the APIs defined in this
+      specification must implement them in a manner consistent with the ECMAScript Bindings
+      defined in the Web IDL specification [[!WEBIDL]], as this specification uses that
+      specification and terminology.
+    </p>
+  </section>
+  <section>
+    <h2>
+      Example
+    </h2>
+    <p>
+      The following example demonstrates a request for multiple display captures using the
+      <code>navigator.mediaDevices.getDisplayMediaSet</code> method defined in this document.
+    </p>
+    <pre class="example highlight">try {
+  let mediaStream = await navigator.mediaDevices.getDisplayMediaSet({video:true});
+  videoElement0.srcObject = mediaStream[0];
+  videoElement1.srcObject = mediaStream[1];
+  // ...
+} catch (e) {
+  console.log('Unable to acquire screen captures: ' + e);
+}</pre>
+  </section>
+  <section id="terminology">
+    <h2>
+      Terminology
+    </h2>
+    <p>
+      This document uses the definition of {{MediaStream}}, {{MediaStreamTrack}}
+      from [[!GETUSERMEDIA]] and <a data-cite="screen-capture#dfn-display-surface">display surface</a>,
+      <a data-cite="screen-capture#dfn-active-user-consent">active user consent</a>,
+      <a data-cite="screen-capture#dfn-floor-value">floor value</a> from [[!screen-capture]].
+    </p>
+  </section>
+  <section id="cmds">
+    <h2>
+      Capturing multiple display surfaces
+    </h2>
+    <p>
+      Capture of multiple <a data-cite="screen-capture#dfn-display-surface">display surfaces</a>
+      is enabled through the addition of a new
+      {{MediaDevices/getDisplayMediaSet}} method on the {{MediaDevices}}
+      interface, that is similar to {{MediaDevices/getUserMedia()}},
+      except that it acquires media from multiple display devices
+      chosen by the end-user each time.
+    </p>
     <section>
-      <h2>Background</h2>
-      <p>TODO</p>
+      <h2>
+        <dfn>MediaDevices</dfn> Additions
+      </h2>
+      <pre class="idl">
+          partial interface MediaDevices {
+            Promise&lt;sequence&lt;MediaStream&gt;&gt; getDisplayMediaSet(optional DisplayMediaStreamOptions options = {});
+          };
+        </pre>
+      <dl data-link-for="MediaDevices" data-dfn-for="MediaDevices" class="methods">
+        <dt>
+          <dfn>getDisplayMediaSet</dfn>
+        </dt>
+        <dd>
+          <p>
+            Prompts the user for permission to live-capture multiple of their
+            <a data-cite="screen-capture#dfn-display-surface">display surfaces</a>.
+          </p>
+          <p>
+            The <a href="#dfn-user-agent">user agent</a> MUST let the end-user choose which
+            <a data-cite="screen-capture#dfn-display-surface">display surfaces</a> to share
+            out of all available choices every time, and MUST NOT use
+            any {{MediaTrackConstraints}} in <var>options</var>.<code>video</code>
```

Comment:
Why is that restriction put in place?
https://w3c.github.io/mediacapture-screen-share/#dfn-displaysurface may be useful.

Reply:
Yes, this spec was in large parts inspired by the getDisplayMedia spec.

My understanding for why this restriction is in place in gDM is to prevent developers into pushing users to share surfaces they would not share otherwise (e.g. removing everything but screen sharing to force user to share screens - and more information - instead of windows or tabs, etc...). I'd argue this restriction also makes a lot of sense for gDMS.